ERCS/erc-7739.md

```diff
@@ -322,37 +315,45 @@ As many developers may not update their applications to support the nested EIP-7
 
 The `typedDataSignTypehash` must be constructed on-the-fly on-chain. This is to enforce that the signed contents will be visible in the signature request, by requiring that `contents` be a user defined type. 
 
-The structure is intentionally made flat with the fields of `eip712Domain` to make implementation feasible. Otherwise, smart accounts must implement on-chain lexographical sorting of strings for the struct type names when constructing `typedDataSignTypehash`.
+The fields of `eip712Domain` are flattened into the `TypedDataSign` structure to enable compatibility with certain Ethers JS versions.
```

Comment:
Can you explain this?

Reply:
Ok I understand.

```suggestion
The fields of `eip712Domain` are flattened into the `TypedDataSign` structure instead of being included as a field of type `EIP712Domain`, because the type could be different from and conflict with the domain type of the verifying contract.
```